From 3dabde1c2d1c6dab26029c9b7cde0b80c8918504 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Wed, 16 Dec 2015 17:02:52 -0500
Subject: [PATCH 1204/1565] drm/amd/dal: get modes in get_modes, instead of
 caching

The old way of populating modes no longer works in the drm-next
kernel tree.

Change-Id: I31e16aabd91db9ce1f2e5086eb3b17e91bbbc76c
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c       | 1 -
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 4 +---
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
index a62c7db..b2b7820 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
@@ -775,7 +775,6 @@ void amdgpu_dm_update_connector_after_detect(
 				(struct edid *) sink->dc_edid.raw_edid;
 			drm_mode_connector_update_edid_property(connector,
 					aconnector->edid);
-			amdgpu_dm_connector_get_modes(&aconnector->base);
 		}
 	} else {
 		drm_mode_connector_update_edid_property(connector, NULL);
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index 07aeedc..8e82acf 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -1237,9 +1237,7 @@ static struct drm_encoder *best_encoder(struct drm_connector *connector)
 
 static int get_modes(struct drm_connector *connector)
 {
-	struct amdgpu_connector *amdgpu_connector =
-			to_amdgpu_connector(connector);
-	return amdgpu_connector->num_modes;
+	return amdgpu_dm_connector_get_modes(connector);
 }
 
 int amdgpu_dm_connector_mode_valid(
-- 
1.9.1

